lib/internal/worker.js

```diff
@@ -259,6 +259,12 @@ class Worker extends EventEmitter {
                          ...new SafeArrayIterator(options.transferList));
 
     this[kPublicPort] = port1;
+    const { port1: toWorkerThread, port2: toHooksThread } = new MessageChannel();
+    if (!isInternal) {
+      // This is not an internal hooks thread => it needs a channel to the hooks thread:
+      // - send it one side of a channel here
+      transferList.push(toHooksThread);
```

Comment:
Does this need to use a primordial? (cc @aduh95)

Reply:
Yes, for everything that touches the ESM loader, we'd want to make it as reliable as possible
```suggestion
      ArrayPrototypePush(transferList, toHooksThread);
```